### PR TITLE
Case study: OpenBazaar

### PR DESCRIPTION
**Pull request by jessicaschilling**
[WIP] Case study series: OpenBazaar. Not ready for merge yet; in-flight document to be used for final editorial review ~~and for reference by @ericronne and/or @akrych so they can see where art would appear in situ~~.

### CODE REVIEW

**Comment by jessicaschilling** [comment]
@autonome and @momack2 -- this is ready for your review ([preview here](https://bafybeieb2wur2wsdrurqz3kmmpf2kwisfriy3bv6eaqdmaoxgmqkmlca24.ipfs.dweb.link/concepts/case-study-openbazaar/)). Note that the following are still missing:
- High-level diagram (trackable in https://github.com/protocol/design-shop/issues/343&sa=D&ust=1595884969218000&usg=AFQjCNEeMPUuhiAoI1MtgQD1XlX3zbUPYQ)
- Final approval from OB1 (want to clear everything on our end first)

Thank you!

**Comment by johnnymatthews** [comment]
I'm gonna convert this PR to a draft while it's still a work-in-progress. Feel free to convert it back to a regular PR once it's ready to merge.

File: docs/concepts/case-study-openbazaar.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+IPFS serves the present-day, short-term storage needs for the OpenBazaar marketplace, but when it comes to longer-term storage, the team is looking forward to also utilizing Filecoin to allow nodes to be compensated for long-term storage. For customers with large storefronts or big customer bases, but who aren’t online all the time, having the potential to offer a third-party storage layer builds a new advantage of the OpenBazaar marketplace into OB1's business model.
+
+Because of the open source-nature of the project, Hoffman can foresee a future where additional services could be also enabled though Filecoin. For example, third parties could offer nodes for redundancy, resilience, backups, and recovery. In short, as both IPFS and Filecoin continue to develop, the OB1 team looks forward to leveraging this evolution.
+
+"We wanted to create OpenBazaar in parallel with where IPFS and Filecoin are moving, and take advantage of that as it evolves," says Hoffman.
+
+_Note: Metrics and other details in this case study are current as of July 2020. Details may change in the interim._
```
Comment (jessicaschilling) [inline docs/concepts/case-study-openbazaar.md:110]:
```suggestion
_Note: Metrics and other details in this case study are current as of August 2020. Details may change in the interim._
```

File: docs/concepts/case-study-openbazaar.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+Even though OpenBazaar is not yet a mainstream shopping platform, it has already proven to be particularly helpful in cases where some products are difficult to come by in the global market. For example, at times Bitcoin mining equipment has been freely available on OpenBazaar when it was difficult to obtain elsewhere. During the COVID-19 pandemic, OpenBazaar was able to get personal protective equipment into people’s hands when other platforms couldn't; N95 masks were at times more widely available on OpenBazaar than via other platforms, and at prices that self-equalized due to the platform's lack of centralized control. Furthermore, this decentralization meant supply and demand dictated price and availability, rather than being subject to an overseeing party determining price or enforcing geographical distribution.
```
Comment (autonome) [inline docs/concepts/case-study-openbazaar.md:26]: Replace "freely available" with something like "easy to acquire"?

File: docs/concepts/case-study-openbazaar.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+### OpenBazaar by the numbers
+
+<NumberBlock :items="[
```
Comment (autonome) [inline docs/concepts/case-study-openbazaar.md:30]: Capitalize value text? eg "Total nodes"

File: docs/concepts/case-study-openbazaar.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+::: callout
+**Protocol Labs is always there when we need them, but we don’t need them very often.**
+
+_&mdash; Brian Hoffman, CEO of OB1_
+:::
+
+Building such a radically new e-commerce model didn't happen without its hiccups, but the OB1 team felt supported all the way. With their migration to IPFS in late 2015, OB1 was one of the first companies to successfully launch a large-scale, fully Web3 platform. "We went through struggles at the start, but working with Protocol Labs was a collaborative effort," says Hoffman. "We worked pretty closely with the core team. The first year or two were challenging, but now things have solidified and we really don’t need much from them." As both IPFS and OpenBazaar have evolved, the consideration remains mutual: At one point, an upgrade to the IPFS protocol was incompatible with OB1's implementation, and the IPFS team rolled back the change in order to allow the two teams to work together on a solution. This willingness to collaborate for the greater good of everyone involved is one of the key reasons OB1 continues to be confident that IPFS is a great choice for OpenBazaar's continued success.
```
Comment (autonome) [inline docs/concepts/case-study-openbazaar.md:57]: s/its hiccups/hiccups/

File: docs/concepts/case-study-openbazaar.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+- Persistent storage of at least a week (to remember buyers' shopping carts and allow for merchants who may not be online 24/7)
+- Support for Bitcoin wallets
+- A customized search engine that crawls the IPFS network to update shop content (the crawler maintains the inventory for OpenBazaar's and Haven's front end)
+- Circuit relay capabilities for working around firewalls and mobile iOS networks
+- The use of `go-ipfs` without IPNS, to optimize for OpenBazaar's extremely large number of product images
+- The use of [`libp2p`](https://libp2p.io/), but with its original addressing scheme
+
+### The architecture
+
+At its highest level, the OpenBazaar network is made out of many nodes, whether desktop/laptop (OpenBazaar) or mobile (Haven). Each node is either a merchant or a seller, with a few exceptions; OB1 does run some gateway nodes and caching nodes themselves. Every node has either the OpenBazaar or Haven app installed, and "inside" the respective app is a complete `go-ipfs` node.
```
Comment (autonome) [inline docs/concepts/case-study-openbazaar.md:86]: s/made out of/made up of/

File: docs/concepts/case-study-openbazaar.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+Another use-case-specific customization was a customized search crawler to comb the network for new content. As merchants add new merchandise, the network needs to display that to the buyers in a timely fashion. OB1's custom crawler searches the network for new content and keeps track of new items in the system so buyers can see and purchase the latest merchandise.
+
+Additionally, OB1 created a TOR router for IPFS, as well as a circuit-relay workaround for firewalls for Haven users on iOS. The ability to create these types of modular add-ins represents a valuable flexibility inherent in IPFS.
+
+## OpenBazaar + IPFS: the future
+
+As OpenBazaar and Haven continue to grow, OB1 is looking at potentially integrating serverless, distributed peer-to-peer database OrbitDB in its IPFS implementation. OpenBazaar and Haven currently use flat files in a directory, which is a bit unusual for a marketplace. Building an e-commerce network on top of flat files isn’t ideal, so an OrbitDB implementation may be a near-term boost.
```
Comment (autonome) [inline docs/concepts/case-study-openbazaar.md:100]: s/integrating serverless/integrating the serverless/

File: docs/concepts/case-study-openbazaar.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+_&mdash; Brian Hoffman, CEO of OB1_
+:::
+
+## Overview
+
+::: right
+<img src="./images/case-studies/logo-openbazaar.png" alt="OpenBazaar logo" width="120">
+:::
+
+[OpenBazaar](https://openbazaar.org/) is a peer-to-peer e-commerce platform where buyers and sellers can participate anonymously and privately without data collection by vendors or any other central authority. The OpenBazaar platform is developed by [OB1](https://ob1.io/), whose projects also include [Haven](https://gethaven.app/), a mobile version of OpenBazaar that offers shopping, chat, and the ability to send cryptocurrencies privately. (The products' codebases are similar, but due to the requirements of some mobile app stores, OB1 needed to bifurcate the two via a separate brand and some code tweaks.)
```
Comment (momack2) [inline docs/concepts/case-study-openbazaar.md:20]:
```suggestion
[OpenBazaar](https://openbazaar.org/) is a peer-to-peer e-commerce platform where buyers and sellers can participate anonymously and privately without data collection by vendors or any other central authority. The OpenBazaar platform is developed by [OB1](https://ob1.io/), whose projects also include [Haven](https://gethaven.app/), a mobile version of OpenBazaar that offers shopping, chat, and the ability to send cryptocurrencies privately. (The products' codebases are similar, but due to the requirements of some mobile app stores, OB1 needed to separate the two with different brands and some code tweaks.)
```

Comment (jessicaschilling) [reply docs/concepts/case-study-openbazaar.md]: Much clearer 😊

File: docs/concepts/case-study-openbazaar.md (hunk 6 above)
Comment (jessicaschilling) [reply docs/concepts/case-study-openbazaar.md:100]: 👍

File: docs/concepts/case-study-openbazaar.md (hunk 5 above)
Comment (jessicaschilling) [reply docs/concepts/case-study-openbazaar.md:86]: 👍

File: docs/concepts/case-study-openbazaar.md (hunk 4 above)
Comment (jessicaschilling) [reply docs/concepts/case-study-openbazaar.md:57]: 👍

File: docs/concepts/case-study-openbazaar.md (hunk 3 above)
Comment (jessicaschilling) [reply docs/concepts/case-study-openbazaar.md:30]: House style for docs is sentence case for everything but the highest-level titles (you can see this in the Audius case study too) 😊

File: docs/concepts/case-study-openbazaar.md (hunk 2 above)
Comment (jessicaschilling) [reply docs/concepts/case-study-openbazaar.md:26]: 👍

**Comment by jessicaschilling** [comment]
Circulating to the OpenBazaar folks for one last check before we push this live.

**Pull request merged**